nano/lib/thread_roles.cpp

```diff
@@ -0,0 +1,136 @@
+#include <nano/lib/thread_roles.hpp>
+#include <nano/lib/utility.hpp>
+
+std::string nano::thread_role::get_string (nano::thread_role::name role)
+{
+	std::string thread_role_name_string;
+
+	switch (role)
+	{
+		case nano::thread_role::name::unknown:
+			thread_role_name_string = "<unknown>";
+			break;
+		case nano::thread_role::name::io:
+			thread_role_name_string = "I/O";
+			break;
+		case nano::thread_role::name::work:
+			thread_role_name_string = "Work pool";
+			break;
+		case nano::thread_role::name::packet_processing:
+			thread_role_name_string = "Pkt processing";
+			break;
+		case nano::thread_role::name::vote_processing:
+			thread_role_name_string = "Vote processing";
+			break;
+		case nano::thread_role::name::block_processing:
+			thread_role_name_string = "Blck processing";
+			break;
+		case nano::thread_role::name::request_loop:
+			thread_role_name_string = "Request loop";
+			break;
+		case nano::thread_role::name::wallet_actions:
+			thread_role_name_string = "Wallet actions";
+			break;
+		case nano::thread_role::name::bootstrap_initiator:
+			thread_role_name_string = "Bootstrap init";
+			break;
+		case nano::thread_role::name::bootstrap_connections:
+			thread_role_name_string = "Bootstrap conn";
+			break;
+		case nano::thread_role::name::voting:
+			thread_role_name_string = "Voting";
+			break;
+		case nano::thread_role::name::signature_checking:
+			thread_role_name_string = "Signature check";
+			break;
+		case nano::thread_role::name::rpc_request_processor:
+			thread_role_name_string = "RPC processor";
```

Comment:
This thread name is very similar to the "RPC process". Suggest something like "RPC req proc".

Reply:
This PR just moves the existing code, so copied this as-is. If a change is needed it should go into another PR to not mix moving and renaming, it's harder to track down in git history later.